warehouse/query-core/src/main/java/datawave/core/iterators/DatawaveFieldIndexCachingIteratorJexl.java

```diff
@@ -921,24 +920,17 @@ else if (key.compareTo(this.lastRangeSeeked.getStartKey()) < 0) {
 
     private void fillSortedSets() throws IOException {
         String sourceRow = this.fiRow.toString();
-        // if we are running fillSortedSets as part of a re-seek after yield, then the fillSet threads would not have
-        // completed when a WaitWindowOverrunException was thrown and therefore the set would not have been marked as
-        // complete when setupRowBasedHdfsBackedSet was called at the top of this method. We will try to copy the
-        // RowBasedHdfsBackedSet from the previously used Ivarator and resume processing.
-        // If this re-seek is from an Ivarator yielding, the startKey will have a yield marker in either the
-        // colFam (for a shard range) or colQual (for a document range)
-        boolean usePreviousSortedSet = WaitWindowObserver.hasMarker(this.lastRangeSeeked.getStartKey());
-        if (usePreviousSortedSet) {
-            if (!resumeFromPreviousCall()) {
-                // if resuming from the previous call fails, then set usePreviousSortedSet=false
-                // to force setup of a new HDFS backed set
-                usePreviousSortedSet = false;
-            }
-        }
-        if (!usePreviousSortedSet) {
+        // If we are running fillSortedSets as part of a re-seek after yield, then the fillSet threads would not have
+        // completed when a WaitWindowOverrunException was thrown. Therefore, the set will not have been marked as
+        // complete when setupRowBasedHdfsBackedSet is called. We will try to copy the RowBasedHdfsBackedSet from
+        // the previously used Ivarator and resume processing.
+        boolean resumeFromIvaratorFutures = resumeFromIvaratorFutures();
```

Comment:
No need to check seekRange startKey first.  If the previously used IvaratorFutures are not found then either that Ivarator wasn't run before or it was completed and we should find the persisted data in the HdfsBackedSortedSet

resumeFromIvaratorFutures is a better name for this variable